pkg/controller/highperf/highperf_controller.go

```diff
@@ -0,0 +1,219 @@
+	})
+
+	return hc
+}
+
+func (c *Controller) Run(numWorkers int, stopCh <-chan struct{}, controllerManagerMetrics *controllersmetrics.ControllerManagerMetrics) {
+	defer utilruntime.HandleCrash()
+
+	ctx, cancelFn := context.WithCancel(context.Background())
+	defer cancelFn()
+	defer c.queue.ShutDown()
+
+	klog.Info("Starting high perf controller")
+	defer klog.Info("Shutting down high perf controller")
+	controllerManagerMetrics.ControllerStarted("highperf")
+	defer controllerManagerMetrics.ControllerStopped("highperf")
+
+	if !cache.WaitForNamedCacheSync("node", stopCh, c.nodeInformer.Informer().HasSynced) {
+		return
+	}
+
+	if !cache.WaitForNamedCacheSync("networks", stopCh, c.nwInformer.HasSynced) {
+		return
+	}
+
+	if !cache.WaitForNamedCacheSync("gkenetworkparamset", stopCh, c.gnpInformer.HasSynced) {
+		return
+	}
+
+	for i := 0; i < numWorkers; i++ {
+		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}
+
+	<-stopCh
+}
+
+func (c *Controller) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *Controller) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+
+	defer c.queue.Done(key)
+
+	c.processNode(ctx, key.(string))
+	// TODO (bmorettodama): handle errors from processNode (try again)
+	c.queue.Forget(key)
+	return true
+}
+
+func (c *Controller) processNode(ctx context.Context, key string) error {
+	node, err := c.nodeInformer.Lister().Get(key)
+
+	if err != nil {
+		klog.Error("Fetching object with key %s from store failed with %v", key, err)
+		return err
+	}
+
+	// check if northInterface and network-status have entries for same network
+	northInterfaces, _ := networkv1.ParseNorthInterfacesAnnotation(node.Annotations[networkv1.NorthInterfacesAnnotationKey])
+	nodeNetworks, _ := networkv1.ParseNodeNetworkAnnotation(node.Annotations[networkv1.NodeNetworkAnnotationKey])
+	var readyNetworks networkv1.NorthInterfacesAnnotation
+
+	// create a list of networks present in both the northInterface and network-status annotations
+	statusMap := make(map[string]string)
+	for _, nodeNetwork := range nodeNetworks {
+		statusMap[nodeNetwork.Name] = ""
+	}
+	for _, northInterface := range northInterfaces {
+		if _, ok := statusMap[northInterface.Network]; ok {
+			readyNetworks = append(readyNetworks, northInterface)
+		}
+	}
+
+	// return when there are no ready netwoks
+	if len(readyNetworks) == 0 {
+		return nil
+	}
+
+	// Fetch network objects and check network type
+	for _, network := range readyNetworks {
+		obj, exists, err := c.nwInformer.GetIndexer().GetByKey(network.Network)
+		if err != nil {
+			klog.Errorf("Fetching %s network object: %v", network.Network, err)
+			return err
+		}
+		if !exists {
+			return nil
+		}
+
+		nwobj := obj.(*networkv1.Network)
+
+		if nwobj.Spec.Type != "Device" {
+			continue
+		}
+
+		klog.Info("Network %s is of type Device", network.Network)
+
+		c.updateConfigMap(nwobj)
```

Comment:
make sure you are only patching the objects once for all networks

Reply:
You mean for that node right? The config map will have an entry for each node, and the resource list will have all networks for that node

Reply:
Fixed by upstream.